File: node.js/lib/member.js
module.exports = Member;

function Member(memberId, memberInfo){
    this.id = memberId;
    if(typeof memberInfo === 'string') {
      this.memberInfo = JSON.parse(memberInfo);
    } else if (typeof memberInfo === 'object') {
      this.memberInfo = memberInfo;
    }
    this.records = {};
    this.recordCount = 0;
    
    this.addRecord = function(record){
        this.records[record.Key]=record;
        this.recordCount++;
    }
    
    this.removeRecord = function(key){
        delete this.records[key];
        this.recordCount--;
    }
    
    this.containsSocket = function(socketId){
        for (key in this.records) {
            if (this.records[key].SocketId === socketId) {
                return true;
            }
        }
        
        return false;
    }
    
    this.containsKey = function(key){
        return undefined!==this.records[key];
    }
    
    this.alive = function(){
      return this.recordCount>0;
    }
}

File: java/src/ly/xstream/streaming/Member.java
package ly.xstream.streaming;

import java.util.*;

public class Member {
	public String id;
	HashMap<String,String> memberInfo;
	
	private HashMap<String,MemberMessage> records = new HashMap<String, MemberMessage>();
	
	Member(String id, HashMap<String,String> memberInfo){
		this.id = id;
	    this.memberInfo = memberInfo;
	}
	
	public Map<String,String> getMemberInfo(){
		return memberInfo;
	}
		
	void addRecord(MemberMessage record){
		records.put(record.item.Key, record);
	}
	
	void removeRecord(String key){
		records.remove(key);
	}
	
	Boolean containsSocket(String socketId){
		for(String key: records.keySet()){
			if(records.get(key).item.SocketId.equals(socketId)){
				return true;
			}
		}
		return false;
	}
	
	Boolean containsKey(String key){
		return records.containsKey(key);
	}
	
	Boolean isAlive(){
		return !records.isEmpty();
	}
}
